### ISSUE TITLE
Combine the various *sync plugins

### ISSUE THREAD

**Issue by xsteadfastx** [sampsyo/beets#1715]
I added the embyupdate plugin (#1714) and @sampsyo had the idea that we maybe could put all the update plugins together. I dont know about the MPD but maybe thats a good idea for the plex and emby one. What do you think?

**Comment by jackwilsdon** [comment]
I think what @sampsyo meant was that instead of "merging" them, we should abstract some of their functionality. I can't see any specific parts of both plexupdate and embyupdate that are the same, however I'm sure either you or @sampsyo have a much better insight on this.

**Comment by sampsyo** [comment]
Yeah, my favorite strategy would be to put as much as possible into a reusable component—but we can still leave them as separate plugins, since the configuration and such is reasonably different. There could be, for example, a BeetsPlugin subclass that just calls you on `cli_exit` if something has changed.

This is related to a more general idea, discussed elsewhere, on having "plugin templates" for plugins that do similar things. For example, a "metadata source" template could help with plugins like discogs, and a "metadata finder" template could help you provide an explicit command as well as an `auto` hook that does the same thing.

**Comment by xsteadfastx** [comment]
The thing is: This sounds like its behind my horizon somehow :) i will read more through the BeetsPlugin class code to understand more of it.
